pysqldb3/geopackage.py

```diff
@@ -521,13 +527,154 @@ def read_gpkg(self, dbo = None, schema = None, table = None, port = 5432, srid =
 
             if not private:
                 try:
-                    dbo.query('grant select on {s}."{t}" to public;'.format(
-                        s=schema, t=output_name),
+                    dbo.query(f'grant select on {schema}."{table}" to public;',
                         timeme=False, internal=True, strict=True)
                 except:
                     pass
             self.rename_geom(dbo, schema, table)
 
+        else:
```

Comment:
this is a lot of code, i would propose moving to a bulk function that is called here